### DISCUSSION TITLE
atomWithStorage - set getOnInit without specifying storage parameter for typed atom

### DISCUSSION THREAD

**Discussion by efie45** [pmndrs/jotai#2413] [Bug report]
I would like to use `atomWithStorage` to persist an array of objects to storage. 

I would like to keep the standard defaults for storage, but also specify the option `getOnInit: true`

However, doing this:

 `const persistedSettings = atomWithStorage<Setting[]>("persistedSettings", [], undefined, { getOnInit: true });`
 
Automatically infers the type of undefined as AsyncStorage and therefore fails on initialization. 

Is there a way to use `{ getOnInit: true }` while keeping everything else as the defaults?

**Comment by dai-shi** [comment]
https://tsplay.dev/m3q6kw
It looks good to me. What do you mean by this 👇 ?
> Automatically infers the type of undefined as AsyncStorage

**Reply by efie45** [reply]
You're right it's working now, my apologies. I'm newish to typescript so I'm not sure what the issue was but I'm assuming I must have had some typescript parser issue with my IDE as the issue has gone away since restarting my machine overnight. I'll go ahead and delete this issue. Thank you for the help!
